gensim/test/test_doc2vec.py

```diff
@@ -280,6 +280,24 @@ def models_equal(self, model, model2):
         self.assertEqual(len(model.docvecs.offset2doctag), len(model2.docvecs.offset2doctag))
         self.assertTrue(np.allclose(model.docvecs.doctag_syn0, model2.docvecs.doctag_syn0))
 
+    def test_discard_model_parameters(self):
+        """Test doc2vec model after discard_model_parameters"""
+        for i in [0, 1]:
+            for j in [0, 1]:
+                model = doc2vec.Doc2Vec(sentences, size=5, min_count=1, hs=i, negative=j)
+                model.discard_model_parameters(remove_doctags_vectors=True)
+                self.assertTrue(len(model['human']), 10)
+                self.assertTrue(model.vocab['graph'].count, 5)
+                if (i == 1):
+                    self.assertTrue(hasattr(model, 'syn1'))
+                else:
+                    self.assertTrue(not hasattr(model, 'syn1'))
+                if (j == 1):
+                    self.assertTrue(hasattr(model, 'syn1neg'))
+                else:
+                    self.assertTrue(not hasattr(model, 'syn1neg'))
+                self.assertTrue(hasattr(model, 'syn0_lockf'))
+
```

Comment:
Seems I "sync' in git without "commit", when I added self.docvecs, 'doctag_syn0' checks :) will fix it